pyvisa_py/protocols/hislip.py

```diff
@@ -0,0 +1,714 @@
+
+MESSAGETYPE: LookupTable = {
+    0: "Initialize",
+    1: "InitializeResponse",
+    2: "FatalError",
+    3: "Error",
+    4: "AsyncLock",
+    5: "AsyncLockResponse",
+    6: "Data",
+    7: "DataEnd",
+    8: "DeviceClearComplete",
+    9: "DeviceClearAcknowledge",
+    10: "AsyncRemoteLocalControl",
+    11: "AsyncRemoteLocalResponse",
+    12: "Trigger",
+    13: "Interrupted",
+    14: "AsyncInterrupted",
+    15: "AsyncMaxMsgSize",
+    16: "AsyncMaxMsgSizeResponse",
+    17: "AsyncInitialize",
+    18: "AsyncInitializeResponse",
+    19: "AsyncDeviceClear",
+    20: "AsyncServiceRequest",
+    21: "AsyncStatusQuery",
+    22: "AsyncStatusResponse",
+    23: "AsyncDeviceClearAcknowledge",
+    24: "AsyncLockInfo",
+    25: "AsyncLockInfoResponse",
+    # reserved for future use         26-127 inclusive
+    # VendorSpecific                  128-255 inclusive
+}
+MESSAGETYPE.update({value: key for (key, value) in MESSAGETYPE.items()})
+
+FATALERRORCODE: LookupTable = {
+    0: "Unidentified error",
+    1: "Poorly formed message",
+    2: "Attempt to use connection without both channels established",
+    3: "Invalid initialization sequence",
+    4: "Server refused connection due to maximum number of clients exceeded",
+    # 5-127:   reserved for HiSLIP extensions
+    # 128-255: device defined errors
+}
+FATALERRORCODE.update({value: key for (key, value) in FATALERRORCODE.items()})
+
+ERRORCODE: LookupTable = {
+    0: "Undefined error",
+    1: "Unrecognized message type",
+    2: "Unrecognized control code",
+    3: "Unrecognized vendor defined message",
+    4: "Message too large",
+    # 5-127:   Reserved
+    # 128-255: Device defined errors
+}
+ERRORCODE.update({value: key for (key, value) in ERRORCODE.items()})
+
+LOCKCONTROLCODE: LookupTable = {
+    0: "release",
+    1: "request",
+}
+LOCKCONTROLCODE.update({value: key for (key, value) in LOCKCONTROLCODE.items()})
+
+LOCKRESPONSECONTROLCODE: LookupTable = {
+    0: "fail",
+    1: "success",
+    2: "successSharedLock",
+    3: "error",
+}
+LOCKRESPONSECONTROLCODE.update(
+    {value: key for (key, value) in LOCKRESPONSECONTROLCODE.items()}
+)
+
+REMOTELOCALCONTROLCODE: LookupTable = {
+    0: "disableRemote",
+    1: "enableRemote",
+    2: "disableAndGTL",
+    3: "enableAndGotoRemote",
+    4: "enableAndLockoutLocal",
+    5: "enableAndGTRLLO",
+    6: "justGTL",
+}
+REMOTELOCALCONTROLCODE.update(
+    {value: key for (key, value) in REMOTELOCALCONTROLCODE.items()}
+)
+
+HEADER_FORMAT = "!2sBBIQ"
+# !  = network order,
+# 2s = prologue ('HS'),
+# B  = message type (unsigned byte),
+# B  = control code (unsigned byte),
+# I  = message parameter (unsigned int),
+# Q  = payload length (unsigned long long)
+HEADER_SIZE = struct.calcsize(HEADER_FORMAT)
+
+
+class Struct(dict):
+    def __init__(self, **kwargs):
+        super(Struct, self).__init__(**kwargs)
+        self.__dict__ = self
+
+
+class Instrument:  # pylint: disable=too-many-instance-attributes
```

Comment:
pylint is not run on this repo so I would prefer to avoid comment specific to it.